Add supports for torchmetrics (#144)
GRAAL-Research/poutyne: docs/source/metrics.rst

.. currentmodule:: poutyne

Poutyne offers three kinds of metrics: batch metrics, epoch metrics and `torch metrics <https://torchmetrics.readthedocs.io/>`__.
The main difference between batch and epoch metrics is that **batch metrics** are computed at each batch and averaged at the end of an epoch whereas **epoch metrics** compute statistics for each batch and compute the metric at the end of the epoch.

> **Torch metrics** can be passed either as epoch metrics (only displayed at the end of the epoch) or using the specialized :class:`torch_metrics argument <poutyne.Model>` (displayed for each batch and at the end of the epoch).

Reviewer: Il manque de contenu sur le choix de passer ça par un nouvel argument je trouve.

> **Torch metrics** can be passed either as epoch metrics (only displayed at the end of the epoch) or using the specialized :class:`torch_metrics argument <poutyne.Model>` (displayed for each batch and at the end of the epoch).

Author: Tu peux pas les passer comme batch metrics, juste epoch ou torch metrics.

.. autoclass:: SKLearnMetrics
    :members:

TorchMetrics
------------

`TorchMetrics <https://torchmetrics.readthedocs.io/>`__ is a library implementing a lot of known metrics in PyTorch.
They are passed to :class:`~poutyne.Model` and :meth:`ModelBundle.from_network() <poutyne.ModelBundle.from_network()>` using either the  ``epoch_metrics`` argument or the ``torch_metrics`` argument.

> When passed as epoch metric, they are only displayed at the end of the epoch.

Reviewer: ```suggestion
When passed as epoch metrics, they are only displayed at the end of the epoch.
```

> When passed using specialized ``torch_metrics`` argument, they are ddisplayed for each batch and at the end of the epoch.

Reviewer: ```suggestion
When passed using the specialized ``torch_metrics`` argument, they are displayed for each batch and at the end of the epoch.
```

See `TorchMetrics documentation <https://torchmetrics.readthedocs.io/en/latest/references/modules.html>`__ for available torch metrics.

> .. warning:: When using this argument, the torch metrics are computed at each batch. This
>     can significantly slow down the computations depending on the metrics used. In such case, we advise to
>     use them as epoch metrics instead.
> 

Reviewer: ```suggestion
.. warning:: When using this argument, the torch metrics are computed at each batch. This
    can significantly slow down the computations depending on the metrics used. In such cases, we advise using them as epoch metrics instead.

```

Reviewer: Due to that warning, I think `model` should raise a warning if a torch metric is passed as a batch metric.

Author: Tu peux pas les passer comme batch metrics, juste epoch ou torch metrics.

Reviewer: donc ça raise un erreur?

Author: L'argument en question ici, c'est `torch_metrics` et non `batch_metrics`.

Author: Tiens, j'ai pushé une modification à ce bout là.

Reviewer: Good for me now.
